projects/core/src/routing/services/url-matcher.service.ts

```diff
@@ -129,6 +124,36 @@ export class UrlMatcherService {
       matcher['_path'] = path; // property added for easier debugging of routes
     }
     return matcher;
+
+    function useFunctionArgument(
+      segments: UrlSegment[],
+      segmentGroup: UrlSegmentGroup,
+      route: Route
+    ) {
+      if (
+        and(
+          route.pathMatch === 'full',
+          or(segmentGroup.hasChildren(), segments.length > 0)
+        )
+      ) {
+        return null;
+      }
+      return { consumed: [], posParams: {} };
+    }
+
+    /**
+     * Logical function to reduce sonar complexity score as matcher scope is limited to inner function.
+     */
+    function or(a: boolean, b: boolean) {
```

Comment:
Logical operations in functions are a less than ideal solution to reducing the sonar score. A better way would be to return the function logic without each condition getting `+2` for being nested in another function but I could not find a way to do this.